hax/hax/configmanager.py

```diff
@@ -0,0 +1,2081 @@
+                      ha_process_events, create_drive_fid, CatalogAdapter,
+                      MotrConsulProcInfo, ha_conf_obj_states, KVAdapter,
+                      MotrConsulProcStatus, MotrProcStatusLocalRemote,
+                      consul_to_local_nodename)
+from hax.exception import HAConsistencyException
+from hax.types import (ByteCountStats, ConfHaProcess, Fid, FsStatsWithTime,
+                       ObjT, ObjHealth, ObjTMaskMap, Profile, PverInfo,
+                       PverState, m0HaProcessEvent, m0HaProcessType,
+                       KeyDelete, HaNoteStruct, m0HaObjState)
+from hax.consul.cache import uses_consul_cache, supports_consul_cache
+
+
+__all__ = ['ConfigManager', 'ConsulConfigManager', 'get_config_manager']
+
+
+LOG = logging.getLogger('hax')
+
+motr_processes_lock = Lock()
+motr_processes_status: dict = {}
+
+
+class ConfigManager(ABC):
+    def __init__(self):
+        self.object_state_getters = {
+            ObjT.SDEV.name: self.get_sdev_state,
+            ObjT.DRIVE.name: self.get_sdev_state,
+            ObjT.NODE.name: self.get_node_state,
+            ObjT.ENCLOSURE.name: self.get_encl_state,
+            ObjT.CONTROLLER.name: self.get_ctrl_state
+        }
+
+    @abstractmethod
+    def kv_get(self, key: str, kv_cache=None,
+               allow_null=False, **kwargs) -> Any:
+        pass
+
+    @abstractmethod
+    def kv_get_raw(self, key: str, **kwargs) -> Tuple[int, Any]:
+        pass
+
+    @abstractmethod
+    def kv_put(self, key: str, data: str, kv_cache=None, **kwargs) -> bool:
+        pass
+
+    @abstractmethod
+    def kv_delete_in_transaction(self, tx_payload: List[KeyDelete]) -> bool:
+        pass
+
+    @abstractmethod
+    def get_node_names(self) -> List[str]:
+        pass
+
+    @abstractmethod
+    def get_service_names(self) -> List[str]:
+        """
+        Return full list of service names currently registered in Consul
+        server.
+        """
+        pass
+
+    @abstractmethod
+    def get_services(self, svc_name: str) -> List[Dict[str, Any]]:
+        """
+        Return service(s) registered in Consul by the given name.
+        """
+        pass
+
+    @abstractmethod
+    def get_local_nodename(self) -> str:
+        """
+        Returns the logical name of the current node. This is the name that
+        Consul is aware of. In other words, whenever Consul references a node,
+        it will use the names that this function can return.
+        """
+        pass
+
+    @abstractmethod
+    def get_session_node(self, session_id: str) -> str:
+        pass
+
+    @abstractmethod
+    def destroy_session(self, session: str) -> None:
+        pass
+
+    @abstractmethod
+    def is_node_alive(self, node: str, kv_cache=None) -> bool:
+        pass
+
+    @abstractmethod
+    def get_node_health_details(
+            self, node: str) -> Optional[List[Dict[str, Any]]]:
+        pass
+
+    def get_process_base_fid(self, proc_fid: Fid) -> Fid:
```

Comment:
This function can be moved to hax/hax/utils.py

Reply:
Sorry, I am not sure, but did we decide to move out Consul specific part in a separate task?

Reply:
Moved this out of the class.

It's somewhat tough to split things up into separate tasks and still maintain functionality

Reply:
Yeah agree, I am okay to make the related changes in the same PR.